azkaban-common/src/main/java/azkaban/executor/ExecutionLogsCleaner.java

```diff
@@ -0,0 +1,69 @@
+package azkaban.executor;
+
+import azkaban.utils.Props;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Singleton
+@SuppressWarnings("FutureReturnValueIgnored")
+public class ExecutionLogsCleaner {
+
+   private static final Logger logger = LoggerFactory.getLogger(ExecutionLogsCleaner.class);
+   private final ScheduledExecutorService scheduler;
+   private final ExecutorLoader executorLoader;
+   private final Props azkProps;
+   private long executionLogsRetentionMs;
+   // 12 weeks
+   private static final long DEFAULT_EXECUTION_LOGS_RETENTION_MS = 3 * 4 * 7
+       * 24 * 60 * 60 * 1000L;
+
+   private static final long DEFAULT_LOG_CLEANUP_INTERVAL_MINUTES = 60;
+   private long cleanupIntervalInMinutes;
+
+   @Inject
+   public ExecutionLogsCleaner(final Props azkProps, final ExecutorLoader executorLoader) {
+      this.azkProps = azkProps;
+      this.executorLoader = executorLoader;
+      this.scheduler = Executors.newSingleThreadScheduledExecutor();
+      this.executionLogsRetentionMs = this.azkProps.getLong("execution.logs.retention.ms",
+          DEFAULT_EXECUTION_LOGS_RETENTION_MS);
+      this.cleanupIntervalInMinutes = this.azkProps.getLong(
+          "execution.logs.cleanup.interval.minutes",
+          DEFAULT_LOG_CLEANUP_INTERVAL_MINUTES);
+   }
+
+   public void start() {
+      logger.info("Starting execution logs clean up thread");
+      this.scheduler.scheduleAtFixedRate(() -> cleanExecutionLogs(), 0L, cleanupIntervalInMinutes,
+          TimeUnit.MINUTES);
+   }
+
+   private void cleanExecutionLogs() {
+      logger.info("Cleaning old logs from execution_logs");
+      final long cutoff = System.currentTimeMillis() - this.executionLogsRetentionMs;
+      logger.info("Cleaning old log files before "
+          + new DateTime(cutoff).toString());
+      cleanOldExecutionLogs(System.currentTimeMillis()
```

Comment:
Can you use the `cutoff` calculated above? If more than one millisecond passes between lines 49 and 52, the log will misstate the cutoff time. (Yes, I know that's how it was before your changes. Still, fix or leave as is?)